File: openacc/lecture201811/vector1.f90
! -*- fortran -*-
program main

  implicit none
  integer :: i, n=10
  double precision :: v1(10), v2(10)

  do i=1, n
     v1(i) = dble(i)
     v2(i) = 0.0d0
  enddo

!$acc kernels copyout(v2(:)) copyin(v1(:))
!$acc loop gang, vector(32)
  do i=1, n
     v2(i) = v1(i) * 2.0d0
  enddo
!$acc end kernels

  do i=1,n
     write(*,'(1H F8.2)',advance="NO")v1(i)
  enddo
  write(*,*)""
  do i=1,n
     write(*,'(1H F8.2)',advance="NO")v2(i)
  enddo
  write(*,*)""

end program main

File: openacc/lecture201811/vector1.c
#include <stdio.h>
#include <stdlib.h>

int main(int argc, char **argv)
{
  int i;
  int n=10;
  double v1[10], v2[10];

  for(i=0; i<n; i++){
	v1[i] = (double)(i+1);
	v2[i] = 0.0;
  }

#pragma acc kernels copyout(v2[:]) copyin(v1[:])
#pragma acc loop gang, vector(32)
  for(i=0; i<n; i++){
	v2[i] = v1[i] * 2.0;
  }

  for(i=0; i<n; i++){ printf(" %.2f", v1[i]); }
  printf("\n");
  for(i=0; i<n; i++){ printf(" %.2f", v2[i]); }
  printf("\n");

  return 0;
}
